### PR TITLE
Prevent duplicate content

### PR DESCRIPTION
**Pull request by daviddavis**
Prevent users/plugins from adding content to a repo version if that content would otherwise already exist for that repo version.

Added validation to the viewset so we don't kick off a task and a nice user error gets return if the content addition/removal isn't allowed. Also added validation to the model mostly for the benefit of plugins who might call these methods directly.

closes #3304
https://pulp.plan.io/issues/3304

### CODE REVIEW

File: pulpcore/pulpcore/app/models/repository.py [hunk 1]
```diff
@@ -373,12 +382,15 @@ def add_content(self, content):
...
             raise ResourceImmutableError(self)
-        with suppress(IntegrityError):
-            association = RepositoryContent(
-                repository=self.repository,
-                content=content,
-                version_added=self)
-            association.save()
+
+        if self.content_exists(content):
+            raise DuplicateContentException(content, self.repository)
```
Comment (jortel) [inline pulpcore/pulpcore/app/models/repository.py]: Do we really want to raise an exception here?  Perhaps add_content() should be idempotent.

File: pulpcore/pulpcore/app/models/repository.py [hunk 2]
```diff
@@ -395,6 +407,9 @@ def remove_content(self, content):
         if self.complete:
             raise ResourceImmutableError(self)
 
+        if not self.content_exists(content):
```
Comment (jortel) [inline pulpcore/pulpcore/app/models/repository.py]: Do we really want to raise an exception here?  Perhaps remove_content() should be idempotent.

**Comment by daviddavis** [comment]
@jortel updated the PR based on your feedback.

**Comment by daviddavis** [comment]
ok test

**Pull request merged**